partner_firstname/models/res_partner.py

```diff
@@ -26,11 +26,30 @@ class ResPartner(models.Model):
         readonly=False,
     )
 
+    @api.model
+    def name_fields_in_vals(self, vals):
+        """Method to check if any name fields are in `vals`."""
+        return vals.get("firstname") or vals.get("lastname")
```

Comment:
I think this method should instead return a list of fields that you could check using:

`set(vals) & self._get_name_fields()`

Or are you doing it this way to consider provided values that are falsy?

Reply:
> Or are you doing it this way to consider provided values that are falsy?

Yes, if neither `firstname` nor `lastname` are set, then they should be computed from the `name` field; otherwise, if any of them are set, `name` should be computed from them instead.